go/vt/sqlparser/ast.go

```diff
@@ -1935,7 +1935,7 @@ type (
 	// SubstrExpr('static string value', value_expression, value_expression)
 	// In this case StrVal will be set instead of Name.
 	SubstrExpr struct {
```

Comment:
I think SubstrExpr is wrong
my understanding is that you can feed it arbitrary expression
```sql
mysql> select substring(1002010,3);
+----------------------+
| substring(1002010,3) |
+----------------------+
| 02010                |
+----------------------+
1 row in set (0.00 sec)
```

here I tried a literal int, that gets coerced into a string

```
mysql> select substring(lower("ALL CAPS"),3);
+--------------------------------+
| substring(lower("ALL CAPS"),3) |
+--------------------------------+
| l caps                         |
+--------------------------------+
1 row in set (0.01 sec)
```

We should remove `StrVal *Literal` and rename `Name` to `Expr`